airflow-core/src/airflow/models/serialized_dag.py

```diff
@@ -412,13 +412,19 @@ def write_dag(
         serialized_dag_hash = session.scalars(
             select(cls.dag_hash).where(cls.dag_id == dag.dag_id).order_by(cls.created_at.desc())
         ).first()
-        dag_version = DagVersion.get_latest_version(dag.dag_id, session=session, load_serialized_dag=True)
+        dag_version = session.scalars(
+            select(DagVersion)
+            .where(
+                DagVersion.dag_id == dag.dag_id,
+                DagVersion.bundle_version == bundle_version,
+                DagVersion.bundle_name == bundle_name,
+            )
+            .options(joinedload(DagVersion.task_instances))
+            .options(joinedload(DagVersion.serialized_dag))
+            .order_by(DagVersion.created_at.desc())
+        ).first()
```

Comment:
```suggestion
        ).limit(1)
```
limit is actually faster

Reply:
You also need to rebase

Reply:
Thanks for the suggestion! I’ll rebase my branch onto origin/main and resolve the conflicts, then update the query as you recommended.